app/frontend/src/features/auth/email/ChangeEmailPage.tsx

```diff
@@ -0,0 +1,95 @@
+import { useMediaQuery, useTheme } from "@material-ui/core";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error as GrpcError } from "grpc-web";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+
+import Alert from "../../../components/Alert";
+import Button from "../../../components/Button";
+import CircularProgress from "../../../components/CircularProgress";
+import PageTitle from "../../../components/PageTitle";
+import TextField from "../../../components/TextField";
+import { service } from "../../../service";
+import useAccountInfo from "../useAccountInfo";
+import useChangeDetailsFormStyles from "../useChangeDetailsFormStyles";
+
+interface ChangeEmailFormData {
+  newEmail: string;
+  currentPassword?: string;
+}
+
+export default function ChangeEmailPage() {
+  const classes = useChangeDetailsFormStyles();
+  const theme = useTheme();
+  const isMdOrWider = useMediaQuery(theme.breakpoints.up("md"));
+
+  const {
+    data: accountInfo,
+    error: accountInfoError,
+    isLoading: isAccountInfoLoading,
+  } = useAccountInfo();
+
+  const {
+    error: changeEmailError,
+    isLoading: isChangeEmailLoading,
+    isSuccess: isChangeEmailSuccess,
+    mutate: changeEmail,
+  } = useMutation<Empty, GrpcError, ChangeEmailFormData>(
+    ({ currentPassword, newEmail }) =>
+      service.account.changeEmail(newEmail, currentPassword)
```

Comment:
I think clearing the form with `reset()` on success would be nice. This applies to all the account-related forms.

Reply:
That wouldn't work though if we want to display the success message without keeping track of that in a separate variable, since clearing on success would also clear that it's succeeded. Not sure I'm too keen on creating a separate variable just for that.

Reply:
I think it will work, you call `reset` from `useForm`, that doesn't effect the mutation right?

Reply:
Oh yeah that would work, sorry got confused since the comment is below the mutation!